### PR TITLE
Fix menu issues

### PR DESCRIPTION
**Pull request by Joe-Brady**
Dorra is aware of the issues below, which are now fixed (by changing dropit.js to work in a different way, and reducing reliance on app.js for the menu system):

- Large screened tablet devices previously could not use the drop-down menu system, because it expected mouse-based input for any windows wider than 899px. Now resolved.

- Mobile-style menu previously opened and closed on mouse hover, which caused usability issues when viewing on a narrow desktop window (< 899px). Now resolved.

- Desktop menu previously disappeared after a series of actions (resize window to mobile width, open and close menu, resize window to desktop width). Now resolved.

### CODE REVIEW

**Comment by charliegracie** [comment]
Hi @Joe-Brady thanks for the contribution.  You will need to sign your commit as outlined here:
https://github.com/eclipse/omr/blob/master/CONTRIBUTING.md

**Comment by Joe-Brady** [comment]
Hi Charlie,

I have used `git commit --amend -s` to sign the original commit.

There are still going to be failing checks because of course this appears as a 'new commit', and the original one will stay there.

Please let me know if this is now acceptable.

**Comment by mstoodle** [comment]
Hi @Joe-Brady , have you signed the Eclipse ECA ? See link under "Legal considerations" in the contribution guidelines Charlie linked you to. Your joe.brady@uk.ibm.com email address needs to be registered with Eclipse as having signed the ECA.

**Comment by Joe-Brady** [comment]
Hi, yes I have - is it not showing up as registered?

**Comment by mstoodle** [comment]
My fault, I didn't noticed you how you had done the update. It would be better if you cleaned up your menu-fixes branch to only have a single, signed commit.  Note that you can rewrite history however you like in your fork using git push -f .

**Comment by Joe-Brady** [comment]
Hi @charliegracie , I have cleaned up the branch into a single signed commit. I also noticed that I now had merge conflicts due to Sue's updates to _includes/header/header-en.html. Because I have completely updated the structure of that file, I have added her simple changes (just a few names of the menu headers) to my version. Is that ok?

I also ended up committing the gemfiles that I used for my local build. Do I need to remove these from my commit?

File: css/style.css [hunk 1]
```diff
@@ -663,7 +663,7 @@ footer {
     float: none;
     border-bottom: 1px solid #ccc;
     margin: 0;
-    min-height: 47px;
+    /*min-height: 47px;*/
```
Comment (mstoodle) [inline css/style.css]: not a big fan of commented out lines: if not needed, please remove

File: css/style.css [hunk 2]
```diff
@@ -704,7 +704,7 @@ footer {
 
   .dropit-trigger > a {
     display: block;
-    margin-bottom: 8px !important;
+    /*margin-bottom: 8px !important;*/
```
Comment (mstoodle) [inline css/style.css]: another commented out line

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
+source "https://rubygems.org"
```
Comment (mstoodle) [inline Gemfile]: is this file required for proper rendering of the web site? If not, I suggest it not be added.

File: Gemfile.lock [hunk 4]
```diff
@@ -0,0 +1,53 @@
+GEM
```
Comment (mstoodle) [inline Gemfile.lock]: is this file required for proper rendering of the web site? if not, I suggest it not be added.

**Review by mstoodle — CHANGES_REQUESTED** [review]

**Comment by Joe-Brady** [comment]
Hi @mstoodle I have removed all commented-out code in styles.css, and removed Gemfile and Gemfile.lock

**Comment by charliegracie** [comment]
Did you get a new version of dropit.js or did you modify it?  

Also your change needs to be rebased on the latest before it can be merged.  You should not add her changes to your change you you should rebase your changes on her changes.  Thanks

**Comment by Joe-Brady** [comment]
@charliegracie , I have now rebased on her changes.

I modified dropit.js because of its (fairly major) flaws. E.g. try using the menu system on a mid-size tablet in landscape mode, when the window width is big enough to trigger the desktop menu layout. It's not possible to access any of the pages, except for the first one in the list.

Let me know if this is all ok now.

**Comment by charliegracie** [comment]
Should be checking in that modified version of the incubating picture?

**Comment by Joe-Brady** [comment]
It's up to you:

- It's a duplicate of the original image, but with the "@2x" suffix.
- There is a JavaScript plugin that looks for double-size images to use when the user is on a high-resolution display, such as the MacBook Pro's 'retina display'.
- It was failing to find the "@2x" version of the incubating image, which was creating an error in both the browser console and the Jekyll build process.
- In the absence of a readily available high-resolution version, I just added a duplicate with the required suffix, to stop the error being generated.
- If you have a high resolution version, we can use that. If not, you can choose whether to keep it there, or remove it and have the error back.

**Comment by charliegracie** [comment]
Ok thanks.  @mstoodle are you ok with the changes now?

**Review by mstoodle — APPROVED** [review]

**Comment by charliegracie** [comment]
I launched the build/deploy job for the new changes.  @Joe-Brady can you verify that the menu now works as you believe.

**Comment by Joe-Brady** [comment]
Hi Charlie, yes, it works as expected.

Thank you!

**Pull request merged**